include/deal.II/meshworker/scratch_data.h

```diff
@@ -0,0 +1,1301 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_meshworker_scratch_data_h
+#define dealii_meshworker_scratch_data_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/differentiation/ad.h>
+
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/mapping_q1.h>
+
+#include <boost/any.hpp>
+
+#include <algorithm>
+#include <map>
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace MeshWorker
+{
+  namespace
+  {
+    /**
+     * Alias used to deduce the OutputType when asking for values or gradients
+     * of shape functions multiplied with the given @p NumberType.
+     */
+    template <int dim, int spacedim, typename NumberType, typename Extractor>
+    using OutputType = typename FEValuesViews::View<dim, spacedim, Extractor>::
```

Comment:
You have this `using` declaration in an anonymous namespace, but that's not a good idea in a header file because that means that you get a new type for each file that `#include`s this file. Declaring types in anonymous namespaces in header files is generally dangerous. Just don't enclose it in an anonymous namespace; if you want to keep it out of the eyes of users, put it into an `internal` namespace.